fetch.bs

```diff
@@ -2172,6 +2243,201 @@ exploits pertain to those <a for=request>types</a>. Also, considering "<code>ima
+
+<p>A <dfn export id=token-binding-id>token binding ID</dfn> is the representation
+of a <a for=/>token binding</a>, as described in
+<a href="https://tools.ietf.org/html/draft-ietf-tokbind-protocol#section-3.2">section 3.2</a>
+of [[TOKBIND-PROTOCOL]].
+
+<p>At a very high level, a user agent negotiates the use of Token Binding with the server when it
+sets up a TLS connection to the server, and saves metadata (the Token Binding protocol version and <a for=/>token binding key parameters</a> resulting from the Token Binding negotiation) for the
+TLS connection. The user agent maintains a <a for=/>token binding key</a> store, where it saves
+different <a for=/>token binding key</a>s to be used with different servers. Whenever the
+<a for=/>token binding key</a> is needed, the user agent looks it up in the
+<a for=/>token binding key</a> store (creating it if needed).
+When the user agent sends a <a for=/>request</a> to the server over the TLS connection,
+it proves possession of the private key by adding and signing a <a for=/>token binding ID</a> in a
+Token Binding HTTP <a for=/>header</a>. The server associates ('binds') credentials that it
+issues to that user agent with a public key in the <a for=/>token binding ID</a>,
+and verifies if bound credentials presented to it by a user agent match the public key in the
+<a for=/>token binding ID</a>.
+
+<p>The `<dfn http-header><code>Sec-Token-Binding</code></dfn>` request <a for=/>header</a>
+is used to transmit token binding metadata from the user agent to the server.
+Its <a for=header>value</a> is a base64url-encoded string [[!RFC4648]]:
+
+<pre>
+Sec-Token-Binding       = 1*( ALPHA / DIGIT / "-" / "_" ) *2( "=" )
+</pre>
+
+<p>The value of the <code>Sec-Token-Binding</code> <a for=/>header</a> is a
+<dfn export id=token-binding-message>Token Binding Message</dfn> as specified in
+<a href="https://tools.ietf.org/html/draft-ietf-tokbind-https#section-2">section 2</a>
+of the Token Binding over HTTP spec [[!TOKBIND-HTTPS]].
+A <a for=/>token binding message</a> is a list of <a for=/>token binding</a>s,
+where each <a for=/>token binding</a> consists of a
+<a for=/>token binding ID</a>, a <a for=/>token binding type</a> and a cryptographic signature.
+
+<h4 id=federated-token-binding>Federated Token Binding</h4>
+
+<p>In the scenario described above, servers bind tokens to be used by user agents
+with those servers.
+To enable existing Web Authorization protocols to use bound credentials,
+there is a need to support scenarios where bound credentials issued by one server
+(e.g., an identity provider) are presented by the user agent to a different server
+(e.g., a relying party).
+[[TOKBIND-PROTOCOL]] and [[TOKBIND-HTTPS]] describe support for Federated Token Binding.
+
+<p>A <dfn export id=referred-token-binding-id>referred token binding ID</dfn> is the
+<a for=/>token binding ID</a> for a server other than the server to which the <a for=/>request</a>
+is being sent.
```

Comment:
What if these servers use the same connection through HTTP/2 connection coalescing?